### ISSUE TITLE
Dev container updates

### ISSUE THREAD

**Issue by bamurtaugh** [microsoft/codespaces-teaching-template-js#18]
Hi there 👋! I'm a PM on the VS Code team working on things like dev containers, and @burkeholland shared this repo and others (https://github.com/microsoft/codespaces-teaching-template-py, https://github.com/microsoft/codespaces-project-template-js, https://github.com/microsoft/codespaces-project-template-py) with me. These look great, and we really appreciate you creating them!

As I was reviewing the dev container configs across these repos, I noticed a few areas to update or optimize to ensure users get the latest and greatest development experience; the areas I mention below apply to this repo along with the others above:

- Update Features syntax: It's awesome you're using Features! It looks like you're using the older syntax for it, so it'd be great to update
     - Example of what these repos used: https://github.com/microsoft/codespaces-teaching-template-js/blob/main/.devcontainer/devcontainer.json#L35
     - Example of updated syntax: https://github.com/devcontainers/images/blob/main/.devcontainer/devcontainer.json#L18
     - The latest Features can be found here: https://github.com/devcontainers/features, and we recently wrote a blog post about using and creating Features: https://code.visualstudio.com/blogs/2022/09/15/dev-container-features
- Reference latest images: As part of work on the open dev container spec (https://containers.dev/), we've moved repos related to dev containers into the [`devcontainers`](http://github.com/devcontainers) GitHub org (as you see above with the devcontainers/features repo)
     - We now use https://github.com/devcontainers/images for our images, rather than what's listed in https://github.com/microsoft/vscode-dev-containers
     - As an example, it looks like https://github.com/microsoft/codespaces-teaching-template-js/blob/main/.devcontainer/Dockerfile#L3 and comments like https://github.com/microsoft/codespaces-project-template-py/blob/main/.devcontainer/Dockerfile#L1 reference the old `vscode-dev-containers` repo
     - The new Node.js image you should use is: https://github.com/devcontainers/images/tree/main/src/javascript-node

Please let me know if you have any questions on the above, or about dev containers or their specification in general. Getting your feedback about our tools and processes is a huge help to our team. Thank you!

**Comment by saragibby** [comment]
Thanks so much for all this amazing information @bamurtaugh! 🙌 
I appreciate the time you took to review the Codespaces repositories we are working on, and provide all the details of what and where to update. 

I made the changes to this repository first, please let me know if you spot anything I missed or other updates that should be made. Thanks again for your help!!

**Comment by bamurtaugh** [comment]
No problem, and thanks for creating this repos and making these changes!

I notice an outstanding reference of vscode-dev-containers in a comment that you'd want to remove: https://github.com/microsoft/codespaces-teaching-template-js/blob/main/.devcontainer/devcontainer.json#L2.

As one other nit: The indentation is also a bit less than we typically use (see https://github.com/devcontainers/images/blob/main/src/javascript-node/.devcontainer/devcontainer.json as an example).

Otherwise I think this is looking great! Thanks again.

**Comment by saragibby** [comment]
Thanks for spotting those items @bamurtaugh! 

I got the comment in devcontainer.json updated and looked at the difference of spacing. In this project I have Prettier set up, with tab width set to 2 spaces, and looks like in devcontainers/images there is a tab width of 8. To stay consistent with the rest of this project I kept the tab width at 2.

Appreciate your help to get these repos dialed in! 😄
